diff -Nur src.orig/Makefile src/Makefile
--- src.orig/Makefile	2010-02-06 02:59:15.000000000 +0100
+++ src/Makefile	2010-02-14 16:16:10.819815931 +0100
@@ -13,6 +13,12 @@
 #
 # $Id: Makefile_kbuild_portsrc,v 1.5.4.1 2009/08/13 21:37:18 Exp $
 
+ifeq ($(shell dpkg --print-architecture),amd64)
+BINARCH = x86_64
+else
+BINARCH = i386
+endif
+
 obj-m              += wl.o
 
 wl-objs            := 
@@ -24,7 +30,7 @@
 EXTRA_CFLAGS       += -I$(src)/src/include
 EXTRA_CFLAGS       += -I$(src)/src/wl/sys -I$(src)/src/wl/phy
 
-EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
+EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped_$(BINARCH)
 
 all:
 	KBUILD_NOPEDANTIC=1 make -C /lib/modules/`uname -r`/build M=`pwd`
